### PR TITLE
Added linter to workflow

### PR DESCRIPTION
**Pull request by aujjwal-redhat**
Added python linting to github workflow to check for typos and other code styling issues.
Updates: #47 

Signed-off-by: aujjwal-redhat <aujjwal@redhat.com>

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+on: [push]
+
+jobs:
+  styles:
+    runs-on: ubuntu-18.04
+    name: Linting
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions/setup-python@v1
+      - uses: whynothugo/python-linting@master
```
Comment (srijan-sivakumar) [inline .github/workflows/linting.yml:12]: what is this whynothugo ? Is it some other repo ?

File: .github/workflows/linting.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: Linting
+
+on: [push]
+
+jobs:
+  styles:
```
Comment (nik-redhat) [inline .github/workflows/linting.yml:6]: rename: `lint-check`

File: .github/workflows/linting.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name: Linting
+
+on: [push]
```
Comment (nik-redhat) [inline .github/workflows/linting.yml:3]: Make it trigger on each `pull_request`

File: .github/workflows/linting.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+jobs:
+  styles:
+    runs-on: ubuntu-18.04
+    name: Linting
```
Comment (nik-redhat) [inline .github/workflows/linting.yml:8]: I don't think this is required.

**Comment by nik-redhat** [comment]
Add an action for `flake` as well to keep the code clean and it will help users keep the formatting correct.

**Comment by srijan-sivakumar** [comment]
> Add an action for `flake` as well to keep the code clean and it will help users keep the formatting correct.

Good idea!

File: .github/workflows/linting.yml (hunk 1 above)
Comment (nik-redhat) [reply .github/workflows/linting.yml:12]: Ohh, there are a lot of custom actions out there. Ig it's one of many, but a better action can be chosen as well.

Comment (aujjwal-redhat) [reply .github/workflows/linting.yml:12]: Yeah for testing

Comment (srijan-sivakumar) [reply .github/workflows/linting.yml:12]: cool..then let's give it a try..it won't hurt as such to try out. We are still exploring all this.

File: .github/workflows/linting.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6.9
```
Comment (nik-redhat) [inline .github/workflows/linting.yml:16]: Above you are saying `Set up Python 3.8`, here you are using version `3.6.9`?

**Review by nik-redhat — COMMENTED** [review]
Have you tested the workflow?

**Review by srijan-sivakumar — APPROVED** [review]

**Comment by srijan-sivakumar** [comment]
I see that we are violating lint rules. LGTM. Let's get this up and running for our PRs.

**Comment by srijan-sivakumar** [comment]
> I see that we are violating lint rules. LGTM. Let's get this up and running for our PRs.

In our existing code..which needs a cleanup but let's focus on that later and separately :p

File: .github/workflows/linting.yml (hunk 5 above)
Comment (nik-redhat) [reply .github/workflows/linting.yml:16]: I think this needs to be updated, though.

**Comment by aujjwal-redhat** [comment]
Rn I am testing the lint on my fork. Will convert this draft for review once done.

File: .github/workflows/linting.yml (hunk 5 above)
Comment (srijan-sivakumar) [reply .github/workflows/linting.yml:16]: Ack..

File: .github/workflows/linting.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8
+          pip install pylint
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          touch __init__.py 
+      - name: Lint with flake8
+        run: |
+          flake8 redant_libs/gluster_libs/peer_ops.py --count --select=E9,F63,F7,F82 --show-source --statistics
+          flake8 redant_libs/gluster_libs/peer_ops.py --count --max-complexity=10 --max-line-length=79 --statistics
```
Comment (nik-redhat) [inline .github/workflows/linting.yml:27]: Why only this file?

File: .github/workflows/linting.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Lint with Pylint
+        run: |
+          pylint redant_libs/gluster_libs/peer_ops.py
```
Comment (nik-redhat) [inline .github/workflows/linting.yml]: Same as above

**Review by nik-redhat — APPROVED** [review]

**Review by srijan-sivakumar — APPROVED** [review]

**Pull request merged**